.github/workflows/tests.yml

```diff
@@ -9,10 +9,11 @@ on:
 jobs:
   TenSEAL:
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
         python-version: [3.6, 3.7, 3.8]
+        os: [windows-latest, macos-latest, ubuntu-latest]
```

Comment:
I don't think a matrix of OSes will work as each platform may differ in the setup phase

Reply:
so I should add each OS separately

Reply:
I would prefer having conditioning on Windows and doing separate build for it, but if that's not possible, removing the matrix might be necessary

Reply:
I can add separate build for windows and mac both.... if it's ok

Reply:
You can add dedicated scripts for each matrix.os.

Something like

```
        run: .github/workflows/scripts/lint_cpp.sh
        if: ${{ matrix.os == 'ubuntu-18.04' }}
```
You can see an example here https://github.com/OpenMined/PSI/blob/2a43e74c07a1eb010006cd5b6621a9435c53a7a9/.github/workflows/tests.yml